README.md

```diff
@@ -1,35 +1,351 @@
+      "displayName": "credit card detector"
+    }
+  }
+  ```
+
+  Within detector struct
+
+  - First specify detectorType as NIGHTFALL_DETECTOR
+  - Pick the nightfall detector you want from the list
+    - API_KEY
+    - CRYPTOGRAPHIC_KEY
+    - RANDOMLY_GENERATED_TOKEN
+    - CREDIT_CARD_NUMBER
+    - US_SOCIAL_SECURITY_NUMBER
+    - AMERICAN_BANKERS_CUSIP_ID
+    - US_BANK_ROUTING_MICR
+    - ICD9_CODE
+    - ICD10_CODE
+    - US_DRIVERS_LICENSE_NUMBER
+    - US_PASSPORT
+    - PHONE_NUMBER
+    - IP_ADDRESS
+    - EMAIL_ADDRESS
+  - Put a display name for your detector, as this will be attached on your findings
+
+- customized regex
+
+  For convenience, we also support customized regex as a detector, which you can fill in as follow.
+
+  ```json
+  {
+    "detector": {
+      "detectorType": "REGEX",
+      "regex": {
+        "pattern": "coupon",
+        "isCaseSensitive": false
+      },
+      "displayName": "simpleRegexCouponDetector"
+    }
+  }
+  ```
+
+- word list
+
+  In case you are not famaliar what regex is, we also support word list searching, simply put in a list of words you want to find.
+
+  ```json
+  {
+    "detector": {
+      "detectorType": "WORD_LIST",
+      "wordList": {
+        "values": ["key", "credential"],
+        "isCaseSensitive": false
+      },
+      "displayName": "simpleWordListKeyDetector"
+    }
+  }
+  ```
+
+- [Extra Parameters Within Detector]
+
+  Besides to define which detector to call, we also allow you to optionally specify some rules you want to attach surround the findinds and finding themselves. They are contextRules and exclusionRules.
+
+  - contextRules
+    A context rule is defined as the surrounding context(pre/post chars) of a finding, you can define a trigger/rule in this section. Once it's triggered, the confidence of finding will be adjusted accordingly base on your confid.
+
+    Example:
+
+    ```json
+    {
+      "detector": {
+        // ...... other detector fileds
+        "contextRules": [
+          {
+            "regex": {
+              "pattern": "test cc",
+              "isCaseSensitive": true
+            },
+            "proximity": {
+              "windowBefore": 30,
+              "windowAfter": 30
+            },
+            "confidenceAdjustment": {
+              "fixedConfidence": "VERY_UNLIKELY"
+            }
+          }
+        ]
+      }
+    }
+    ```
+
+    - regex field specifies a regex to trigger
+    - proximity specifies how many pre|post chars surround findings we want to do the search
+    - confidenceAdjustment specifies what's the confidence you want to change for the finding once trigger the rule
+
+    In this example, if we have a real text like test cc: 4242-4242-4242-4242, and 4242-4242-4242-4242 is detected as a credit card number with confidence of POSSIBLE. After we applied such context rules, since the pre chars test cc matches the regex, the confidence of such findings will drop down to VERY_UNLIKELY as specified
```

Comment:
```suggestion
    As an example, say we have the following line of text in a file `test cc: 4242-4242-4242-4242`, and `4242-4242-4242-4242` is detected as a credit card number finding with confidence level `POSSIBLE`. If we had the context rule above, the confidence level of this finding will be bumped up to `VERY_LIKELY` because the characters preceding the finding,  `test cc`, match the regex.
```

Reply:
same my cc here